packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m

```diff
@@ -86,6 +87,122 @@ - (void)testHandleResultTileDownsamplesWideGamutImages {
+                 "Camera zoom should be updated to 10.0 after animation.");
+}
+
+- (void)testAnimateCameraWithUpdateAndDuration {
+  NSObject<FlutterPluginRegistrar> *registrar = OCMProtocolMock(@protocol(FlutterPluginRegistrar));
+
+  CGRect frame = CGRectMake(0, 0, 100, 100);
+  GMSMapViewOptions *mapViewOptions = [[GMSMapViewOptions alloc] init];
+  mapViewOptions.frame = frame;
+
+  // Init camera with zero zoom.
+  mapViewOptions.camera = [[GMSCameraPosition alloc] initWithLatitude:0 longitude:0 zoom:0];
+
+  PartiallyMockedMapView *mapView = [[PartiallyMockedMapView alloc] initWithOptions:mapViewOptions];
+
+  FLTGoogleMapController *controller =
+      [[FLTGoogleMapController alloc] initWithMapView:mapView
+                                       viewIdentifier:0
+                                   creationParameters:[self emptyCreationParameters]
+                                            registrar:registrar];
+
+  NSArray *zoomToArray = @[ @"zoomTo", @(10.0) ];
+  FGMPlatformCameraUpdate *cameraUpdate = [FGMPlatformCameraUpdate makeWithJson:zoomToArray];
+
+  // Define a configuration with a animation duration of 100ms
+  FGMPlatformCameraUpdateAnimationConfiguration *config =
+      [FGMPlatformCameraUpdateAnimationConfiguration makeWithDurationMilliseconds:@(100)];
+
+  FlutterError *error = nil;
+
+  id mockDartCallbackHandler = OCMClassMock([FGMMapsCallbackApi class]);
+
+  controller.dartCallbackHandler = mockDartCallbackHandler;
+
+  XCTestExpectation *cameraIdleExpectation =
+      [self expectationWithDescription:@"Camera becomes idle"];
+
+  OCMStub([mockDartCallbackHandler didIdleCameraWithCompletion:[OCMArg invokeBlock]])
+      .andDo(^(NSInvocation *invocation) {
+        // Fulfill the expectation when the camera idle callback is invoked
+        [cameraIdleExpectation fulfill];
+      });
+
+  // Trigger the camera animation with configuration
+  [controller.callHandler animateCameraWithUpdate:cameraUpdate
+                                 andConfiguration:config
+                                            error:&error];
+
+  XCTAssertNotEqual(controller.mapView.camera.zoom, 10.0,
+                    "Camera zoom should not be updated immediately.");
+
+  // Wait for the camera idle event with a timeout of 150ms.
+  // This timeout is shorter than the default animation duration (~300ms) to verify that the
+  // specified configuration duration is being applied.
+  [self
+      waitForExpectationsWithTimeout:0.15
+                             handler:^(NSError *_Nullable error) {
+                               if (error) {
+                                 XCTFail(@"Failed to receive camera idle event in time: %@", error);
+                               }
+                             }];
+
+  // Assert that the camera zoom is updated after the animation completes
+  XCTAssertEqual(controller.mapView.camera.zoom, 10.0,
+                 "Camera zoom should be updated to 10.0 after animation.");
+}
```

Comment:
Actually running animations is pretty slow for unit tests; could we just dependency inject a simple wrapper around `CATransaction` and ensure that's being called as expected, rather than testing the animation itself (which is largely testing the underlying SDK rather than the plugin code)?